packages/beacon-node/src/chain/beaconProposerCache.ts

```diff
@@ -7,19 +7,28 @@ const PROPOSER_PRESERVE_EPOCHS = 2;
 
 export type ProposerPreparationData = routes.validator.ProposerPreparationData;
 
+/** Cache to track the validators registored with beacon node. Could eventually be promoted as a
+ * full monitor
+ */
 export class BeaconProposerCache {
+  /** Flag to switch on/off validator stats default set to true */
+  readonly validatorMonitor: boolean;
```

Comment:
recommend removing for now

Reply:
you can repurpose the flag later on if required, whats the point of removing it since this is what you have been asking for?

Reply:
can keep it here, mostly about the CLI flag. I am not sure having a flag which enabled validator monitoring metrics is correct approach

Reply:
removed